src/azure-cli-testsdk/azure/cli/testsdk/base.py

```diff
@@ -177,68 +98,13 @@ def create_random_name(self, prefix, length):
             name = create_random_name(prefix, length)
             self.name_replacer.register_name_pair(name, moniker)
             return name
-
-        return moniker
-
-    def _process_request_recording(self, request):
-        if self.in_recording:
-            for processor in self.recording_processors:
-                request = processor.process_request(request)
-                if not request:
-                    break
         else:
-            for processor in self.replay_processors:
-                request = processor.process_request(request)
-                if not request:
-                    break
+            return moniker
```

Comment:
Remove this `else`, try to respect as close as possible the initial code if this is not an improvement but just detail, to simplify review.